### PR TITLE
[LM-10572] Context Prop Management

### PR DESCRIPTION
**Pull request by jaixchaudhary**

### CODE REVIEW

File: example/App.tsx [hunk 1]
```diff
@@ -27,33 +22,40 @@ import {
...
 
 const Stack = createNativeStackNavigator();
 
 function App(): React.JSX.Element {
-  const userName = '';
-  const userUniqueId = '';
-  const chatroomId = '';
+  const userName = 'Jai';
+  const userUniqueId = 'Jai';
+  const chatroomId = '4033778';
```
Comment (jaixchaudhary) [inline example/App.tsx:32]: remove default values

File: likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx [hunk 2]
```diff
@@ -0,0 +1,2105 @@
...
+
+  const handleModalClose = () => {
+    setModalVisible(false);
+  };
+
+  const handleReactionModalClose = () => {
+    setIsReact(false);
+  };
+
+  const leaveChatroom = async () => {
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx:927]: add comments to these methods

File: likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx [hunk 3]
```diff
@@ -0,0 +1,2105 @@
...
+
+  const showWarningModal = () => {
+    setIsWarningMessageModalState(true);
+  };
+
+  const hideWarningModal = () => {
+    setIsWarningMessageModalState(false);
+  };
+
+  const leaveSecretChatroom = async () => {
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx:991]: add comments for these methods

File: likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx [hunk 4]
```diff
@@ -0,0 +1,2105 @@
...
+          }, 300);
+        }
+      })
+      .catch(() => {
+        Alert.alert("Leave Chatroom failed");
+      });
+    return res;
+  };
+
+  const joinChatroom = async () => {
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx:1048]: add comments to all the methods below

File: likeminds-chat-reactnative-integration/ChatSX/customModals/ChatroomActionModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                    } else if (val?.id === ChatroomActions.UNBLOCK_MEMBER) {
+                      // await unblockMember();
+                      setModalVisible(false);
+                    } else if (val?.id === ChatroomActions.SHARE) {
+                      //   Share flow
+                      //   onShare(
+                      //     chatroomID,
+                      //     chatroomType,
+                      //     chatroomDBDetails?.isSecret
+                      //   );
```
Comment (arnavr15) [inline likeminds-chat-reactnative-integration/ChatSX/customModals/ChatroomActionModal/index.tsx:84]: not understood why muteNotifications(), unblockMember(), onShare() etc are commented out

File: likeminds-chat-reactnative-integration/ChatSX/context/MessageListContext.tsx [hunk 6]
```diff
@@ -0,0 +1,762 @@
...
+  const [shouldLoadMoreChatStart, setShouldLoadMoreChatStart] = useState(true);
+  const [response, setResponse] = useState<any>([]);
+  const [flashListMounted, setFlashListMounted] = useState(false);
+  const [isFound, setIsFound] = useState(false);
+
+  const flatlistRef = useRef<any>(null);
+  const dispatch = useAppDispatch();
+  const { messageSent }: any = useAppSelector((state) => state.chatroom);
+  const myClient = Client.myClient;
+  // const { user } = useAppSelector((state) => state.homefeed);
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/context/MessageListContext.tsx:92]: remove this unnecessary comment

File: likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx [hunk 7]
```diff
@@ -0,0 +1,2105 @@
...
+        );
+      })
+      .catch(() => {
+        Alert.alert("Join Chatroom failed");
+      });
+
+    return res;
+  };
+
+  const joinSecretChatroom = async () => {
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx:1093]: add comments for this method

File: likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx [hunk 8]
```diff
@@ -0,0 +1,2105 @@
...
+        setMsg("Notifications muted for this chatroom");
+        setIsToast(true);
+      })
+      .catch(() => {
+        Alert.alert("Mute Notification failed");
+      });
+  };
+
+  const unmuteNotifications = async () => {
+    const payload = {
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx:1165]: add comments to these methods

File: likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx [hunk 9]
```diff
@@ -0,0 +1,2105 @@
...
+
+  const showDMBlockAlert = () => {
+    setDMBlockAlertModalVisible(true);
+  };
+
+  const hideDMBlockAlert = () => {
+    setDMBlockAlertModalVisible(false);
+  };
+
+  const uploadResource = async ({
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/context/ChatroomContext.tsx:1708]: add comments to all the methods, check accordingly

**Pull request closed without merging**